lab-derek/test/test-index.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const superagent = require('superagent');
+const expect = require('expect');
+
+const server = require('../lib/server.js');
+
+//TODO: your tests should start your server when they begin and stop your server when they finish
+
+describe('testing task routes', () => {
+  before((done) => {
+    server.listen(3000, () => {
+      console.log('server up for testing');
+      done();
+    });
+  });
+  after((done) => {
+    server.close(() => {
+      console.log('testing finished, server closed');
+      done();
+    });
+  });
+  //TODO: write a test to ensure that your api returns a status code of 404 for routes that have not been registered
+  it('should return a status code 404 for unregistered routes', () => {
+    superagent.post('localhost:3000/test')
+      .send()
+      .end((err, res) => {
+        expect(res.status).toEqual(400);
+        done();
+      });
+  });
+
+
+  //Write tests to ensure your /api/tasks/simple-resource-name endpoint responds as described for each condition below:
+
+  //TODO: GET - test 404, responds with 'not found' for valid request made with an id that was not found
+
+  it('should respond with "not found" for valid requests with unknown id', () => {
+    superagent.get('/api/tasks?id=test')
+      .end((err, res) => {
+        expect(res.status).toEqual(404);
+        expect(res.body).toEqual('not found');
+        done();
+      });
+  });
+
+
+  //TODO: GET - test 200, response body like {<data>} for a request made with a valid id
+  //TODO: POST - test 201, response body like {<data>} for a post request with a valid body
+
+  it('should POST a new object and return it, then return a status code 200 and valid data if valid id is provided in query string', () => {
+    superagent.post('localhost:3000/api/tasks')
+      .send(
+        {
+          "taskName": "test","xp": "50","id": "10",
+        }
+      ).end((err, res) => {
+        expect(res.status).toEqual(201);
+        expect(res.body).toEqual(
+          {
+            "taskName": "test","xp": "50","id": "10",
+          }
+        );
+      });
+    superagent.get('localhost:3000/api/tasks?=10')
+      .end((err, res) => {
+        expect(res.status).toEqual(200);
+        expect(res.body).toEqual(
+          {
+            "taskName": "test","xp": "50","id": "10",
+          });
+        done();
```

Comment:
I would break this out into two blocks. If you are creating a POST so you can test the GET, consider moving the POST request into a before block. Then it will make that requests before the 'it' block even runs.